platforms/nuttx/Debug/launch.json.in

```diff
@@ -21,6 +21,24 @@
                 "set print pretty",
             ]
         },
+        {
+            "name": "stlink (@PX4_BOARD@)",
+            "device": "@DEBUG_DEVICE@",
+            "svdFile": "@DEBUG_SVD_FILE_PATH@",
+            "executable": "${command:cmake.launchTargetPath}",
+            "request": "launch",
+            "type": "cortex-debug",
+            "servertype": "stlink",
```

Comment:
```suggestion
            "servertype": "stutil",
```
I actually used `"stutil"`.  This requires either a working version of `stlink-tools` installed (not sure if any work reliably), or to build and install directly from https://github.com/stlink-org/stlink.

The poorly-named `stutil` application starts a GDB server, and VSCode seems to understand how to use it from there.